Twitchdeck.Test/UserInterfaceTests.fs

```diff
@@ -26,23 +26,18 @@ let stackPanelChildren view =
     |> attr<ViewElement> "Content"
     |> attr<ViewElement[]> "Children"
 
-// THIS DOES NOT WORK - FIXME
 let rec descendents (view: ViewElement) : ViewElement list =
     let (content : ValueOption<ViewElement>) = view.TryGetAttribute "Content"
-    let contentList =
-        match content with 
-        | ValueSome x -> [x] @ descendents x
-        | ValueNone -> []
-
     let (children : ValueOption<ViewElement[]>) = view.TryGetAttribute "Children"
-    let childrenList =
-        match children with 
-        | ValueSome x ->
-            List.ofArray x
-            |> List.collect (fun x -> descendents x)
-        | ValueNone -> []
-
-    contentList @ childrenList
+    [ match content with
+      | ValueSome content' -> yield content'
+      | ValueNone -> ()
+
+      match children with
+      | ValueSome children' ->
+          for child in children' do
+              yield! descendents child
```

Comment:
Just so that I'm clear on the use of `yield!` - is it essentially a yield monadic bind?

Reply:
`yield!` merges something that returns a sequence into another sequence

e.g.

```
let list1 = [1; 2; 3]
let list2 = [ yield! list1 ]

// now list 2 = [1; 2; 3], it's a shortcut for `yield 1; yield 2; yield 3`